byway/src/main/webapp/track-interests.js

```diff
@@ -58,13 +54,33 @@ function updateStatus(place, elem) {
   }
 }
 
+/**
+ * Finds the list of interests that were set for the trip.
+ * If it is a new trip, no interests are listed. If the trip
+ * had previously-saved interests, return them.
+ * @return fulfilled promise containing the list of interests
+ */
+function getTripInterests() {
+  return new Promise(resolve => {
+    let tripInterests = new Set();
+    fetchInterests(tripKey)
+    .then((trip) => {
+      for (let interest of trip.interests) {
+        tripInterests.add(interest);
+      }
+      resolve(tripInterests);
```

Comment:
Might be useful to have a reject(new Error ...) which you can catch in your function await function call.

Reply:
I don't think any reject is necessary here since it is just loading/saving the data from the backend. If the trip.interests is empty/null, it would resolve null and return an empty Set. I could add a check for the reject case, but that would imply there was an actual error with the Error argument, when it really just means it's a new trip with no interests yet. 

I don't think every Promise needs to handle both cases, so I think I will leave this as is since there is no real "error" to reject.